src/test/methods/batch-write.ts

```diff
@@ -0,0 +1,206 @@
+	updateStub.withArgs(throttleFixture).onFirstCall().returns(stubPromise(throttleData));
+	updateStub.withArgs(throttleFixture).onSecondCall().returns(stubPromise(throttleData));
+	updateStub.withArgs(throttleFixture).onThirdCall().returns(stubPromise(throttleData));
+	updateStub.withArgs(throttleFixture).returns(stubPromise(response));
+	updateStub.returns(stubPromise(response));
+});
+
+test.after(() => {
+	sandbox.restore();
+});
+test.serial('throws error if no items are passed', async t => {
+	await t.throwsAsync(() => db.batchWrite([]).exec(), 'Items array can not be empty.');
+});
+test.serial('throws error if more than 25 items are passed', async t => {
+	const batchItems = [];
+	for (let i = 0; i < 26; i++) {
+		// @ts-ignore
+		batchItems.push(new PutRequest('test', i, i));
+	}
+	await t.throwsAsync(() => db.batchWrite(batchItems).exec(), 'Can not insert more than 25 items at a time.');
+});
+test.serial('insert batch', async t => {
+	await db.batchWrite(
+		[
+			Table1.createBatchPutItem(
+				{partitionKey: 'PK', sortKey: 'SK'},
+				{name: 'name', lastname: 'lastname'}
+				),
+			Table1.createBatchPutItem(
+				{partitionKey: 'PK', sortKey: 'SK23'},
+				{name: 'name23', lastname: 'lastname23'}
+			),
+			Table2.createBatchDeleteItem(
+				{partitionKey: 'PK2', sortKey: 'SK2'}
+				),
+			Table2.createBatchDeleteItem(
+				{partitionKey: 'PK2', sortKey: 'SK3'}
+			),
+			Table2.createBatchPutItem(
+				{partitionKey: 'PK', sortKey: 'SK'},
+				{name: 'name', lastname: 'lastname'}
+			)
+		]
+	).exec();
+	t.deepEqual(updateStub.lastCall.args[0], {
+			RequestItems: {
+				'insert-Table1': [
+					{
+						PutRequest: {
+							Item: {
+								partitionKey: 'PK',
+								sortKey: 'SK',
+								name: 'name',
+								lastname: 'lastname'
+							}
+						}
+					},
+					{
+						PutRequest: {
+							Item: {
+								partitionKey: 'PK',
+								sortKey: 'SK23',
+								name: 'name23',
+								lastname: 'lastname23'
+							}
+						}
+
+					}
+				],
+				'insert-Table2': [
+					{
+						DeleteRequest: {
+							Key:
+								{partitionKey: 'PK2', sortKey: 'SK2'}
+						}
+					},
+					{
+						DeleteRequest: {
+							Key:
+								{partitionKey: 'PK2', sortKey: 'SK3'}
+						}
+					},
+					{
+						PutRequest: {
+							Item: {
+								partitionKey: 'PK',
+								sortKey: 'SK',
+								name: 'name',
+								lastname: 'lastname'
+							}
+						}
+					}
+				]
+			}
+		}
+	);
+});
+test.serial('program should use exponential backoff', async t => {
+	await db.batchWrite(
+		[
+			Table1.createBatchPutItem(
+				{partitionKey: 'PK', sortKey: 'SK'},
+				{name: 'name', lastname: 'lastname'}
+			)
+		]
+	).exec();
+	t.deepEqual(updateStub.lastCall.args[0], {
+			RequestItems: {
+				'insert-Table1': [
+					{
+						PutRequest: {
+							Item: {
+								partitionKey: 'PK',
+								sortKey: 'SK',
+								name: 'name',
+								lastname: 'lastname'
+							}
+						}
+					}
+				]
+			}
+		}
+	);
```

Comment:
How does this test actually test that it has beem retried? We could check `updateStub.callCount` before and after we call the batchWrite for example

Reply:
I agree, I will extend this test to properly test the exponential backoff.

Reply:
The test will now check that every functional call has happened and was called the correct arguments.